core/src/test/java/org/elasticsearch/action/admin/indices/create/CreateIndexWaitsIT.java

```diff
@@ -0,0 +1,167 @@
+                                               .put(indexSettings())
+                                               .put(INDEX_NUMBER_OF_SHARDS_SETTING.getKey(), randomIntBetween(2, 5))
+                                               .put(INDEX_NUMBER_OF_REPLICAS_SETTING.getKey(), 0);
+        if (internalCluster().getNodeNames().length > 0) {
+            String exclude = String.join(",", internalCluster().getNodeNames());
+            settingsBuilder.put("index.routing.allocation.exclude._name", exclude);
+        }
+        Settings settings = settingsBuilder.build();
+        assertThrows(prepareCreate(indexName)
+                         .setSettings(settings)
+                         .setWaitForActiveShards(randomBoolean() ? ActiveShardCount.ONE : ActiveShardCount.ALL)
+                         .setTimeout("2s"), ElasticsearchTimeoutException.class);
+    }
+
+    public void testCreateIndexNoActiveShardsNoWaiting() throws Exception {
+        final String indexName = "test-idx";
+        Settings.Builder settingsBuilder = Settings.builder()
+                                               .put(indexSettings())
+                                               .put(INDEX_NUMBER_OF_SHARDS_SETTING.getKey(), randomIntBetween(2, 5))
+                                               .put(INDEX_NUMBER_OF_REPLICAS_SETTING.getKey(), 0);
+        if (internalCluster().getNodeNames().length > 0) {
+            String exclude = String.join(",", internalCluster().getNodeNames());
+            settingsBuilder.put("index.routing.allocation.exclude._name", exclude);
+        }
+        Settings settings = settingsBuilder.build();
+        CreateIndexResponse response = prepareCreate(indexName)
+                                           .setSettings(settings)
+                                           .setWaitForActiveShards(ActiveShardCount.from(0))
+                                           .setTimeout("2s")
+                                           .get();
+        assertTrue(response.isAcknowledged());
+    }
+
+    public void testCreateIndexNotEnoughActiveShardsTimesOut() throws Exception {
+        final String indexName = "test-idx";
+        final int numReplicas = randomIntBetween(2, 4);
+        Settings settings = Settings.builder()
+                                .put(indexSettings())
+                                .put(INDEX_NUMBER_OF_SHARDS_SETTING.getKey(), randomIntBetween(2, 7))
+                                .put(INDEX_NUMBER_OF_REPLICAS_SETTING.getKey(), numReplicas)
+                                .build();
+        final int numNodes = randomIntBetween(1, numReplicas - 1);
+        internalCluster().ensureAtMostNumDataNodes(numNodes);
+        assertThrows(prepareCreate(indexName)
+                         .setSettings(settings)
+                         .setWaitForActiveShards(ActiveShardCount.from(numNodes + 1))
+                         .setTimeout("2s"), ElasticsearchTimeoutException.class);
```

Comment:
having the tests wait 2 seconds is way too long. Please increase the timeouts to 1minute (or leave default) when we expect NOT to have timeouts, otherwise set timeout to 0 so that we instantly run into timeout.